cpp/tests/unary/unary_ops_test.cu

```diff
@@ -55,6 +55,155 @@ cudf::test::fixed_width_column_wrapper<T> create_expected_columns(cudf::size_typ
     }
 }
 
+template <typename T>
+struct cudf_logical_test : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(cudf_logical_test, cudf::test::NumericTypes);
+
+TYPED_TEST(cudf_logical_test, LogicalNot)
+{
+    cudf::size_type colSize = 1000;
+    std::vector<TypeParam> h_input_v(colSize, false);
+    std::vector<cudf::experimental::bool8> h_expect_v(colSize);
+
+    std::transform(
+        std::cbegin(h_input_v),
+        std::cend(h_input_v),
+        std::begin(h_expect_v),
+        [] (TypeParam e) -> cudf::bool8 {
+            return static_cast<cudf::bool8>(!e);
+        });
+
+    cudf::test::fixed_width_column_wrapper<TypeParam>                 input    (std::cbegin(h_input_v),  std::cend(h_input_v));
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected (std::cbegin(h_expect_v), std::cend(h_expect_v));
+
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::NOT);
+
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_logical_test, SimpleLogicalNot)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam>                 input    {{ true,  true,  true,  true  }};
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected {{ false, false, false, false }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::NOT);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+template <typename T>
+struct cudf_math_test : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(cudf_math_test, cudf::test::NumericTypes);
+
+TYPED_TEST(cudf_math_test, ABS)
```

Comment:
Curious, why are ABS and SQRT are tested differently than all other ops?

Reply:
These functions (with the input I set up) don't involve truncation - if I used others I might get a rounding error and for some the functions fail compilation due `narrowing conversion`